internal/metrics/interceptors_test.go

```diff
@@ -0,0 +1,121 @@
+	r.message = "hello"
+	return t.invokeResult
+}
+
+func newTestCC() *grpc.ClientConn {
+	testCC, _ := grpc.Dial("dns:localhost:9999", grpc.WithInsecure())
+	return testCC
+}
+func Test_GRPCClientInterceptor(t *testing.T) {
+
+	interceptor := GRPCClientInterceptor("test_service")
+
+	tests := []struct {
+		name                          string
+		method                        string
+		errorCode                     error
+		wantgrpcClientResponseSize    string
+		wantgrpcClientRequestDuration string
+		wantgrpcClientRequestCount    string
+	}{
+		{
+			name:                          "ok authorize",
+			method:                        "/authorize.Authorizer/Authorize",
+			errorCode:                     nil,
+			wantgrpcClientResponseSize:    "{ { {grpc_service authorize.Authorizer}{host dns:localhost:9999}{method Authorize}{service test_service}{status OK} }&{1 5 5 5 0 [0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]",
+			wantgrpcClientRequestDuration: "{ { {grpc_service authorize.Authorizer}{host dns:localhost:9999}{method Authorize}{service test_service}{status OK} }&{1",
+			wantgrpcClientRequestCount:    "{ { {grpc_service authorize.Authorizer}{host dns:localhost:9999}{method Authorize}{service test_service}{status OK} }&{1",
+		},
+		{
+			name:                          "unknown validate",
+			method:                        "/authenticate.Authenticator/Validate",
+			errorCode:                     status.Error(14, ""),
+			wantgrpcClientResponseSize:    "{ { {grpc_service authenticate.Authenticator}{host dns:localhost:9999}{method Validate}{service test_service}{status Unavailable} }&{1 5 5 5 0 [0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]",
+			wantgrpcClientRequestDuration: "{ { {grpc_service authenticate.Authenticator}{host dns:localhost:9999}{method Validate}{service test_service}{status Unavailable} }&{1",
+			wantgrpcClientRequestCount:    "{ { {grpc_service authenticate.Authenticator}{host dns:localhost:9999}{method Validate}{service test_service}{status Unavailable} }&{1",
+		},
+		{
+			name:                          "broken method parsing",
+			method:                        "f",
+			errorCode:                     status.Error(14, ""),
+			wantgrpcClientResponseSize:    "{ { {host dns:localhost:9999}{service test_service}{status Unavailable} }&{1 5 5 5 0 [0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]",
+			wantgrpcClientRequestDuration: "{ { {host dns:localhost:9999}{service test_service}{status Unavailable} }&{1",
+			wantgrpcClientRequestCount:    "{ { {host dns:localhost:9999}{service test_service}{status Unavailable} }&{1",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			view.Unregister(grpcViews...)
+			view.Register(grpcViews...)
+
+			invoker := testInvoker{
+				invokeResult: tt.errorCode,
+			}
+			var reply testProto
+
+			interceptor(context.Background(), tt.method, nil, &reply, newTestCC(), invoker.UnaryInvoke)
+
+			// grpcClientResponseSize
+			data, _ := view.RetrieveData(grpcClientResponseSize.Name())
```

Comment:
I know it's just tests but prefer to check err condition with`t.Fatalf`

Reply:
Yes...I actually am pushing a generalized version of those tests, which includes this check.  Can you give that a look vs the previous (commented out) implementation?  It should work for the http metrics too.

Reply:
Should be all set.